spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/caffe/CaffePersister.scala

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.utils.caffe
+
+import java.io._
+
+import scala.collection.JavaConverters._
+import caffe.Caffe.{LayerParameter, NetParameter, V1LayerParameter}
+import com.intel.analytics.bigdl.nn.{Container, Graph, Sequential, View}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Node
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+import caffe.Caffe
+import com.google.protobuf.{CodedOutputStream, GeneratedMessage, TextFormat}
+import com.intel.analytics.bigdl.nn.Graph.ModuleNode
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.tensor.Tensor
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.io.IOUtils
+import org.apache.log4j.Logger
+/**
+ * An utility to convert BigDL model into caffe format and persist into local/hdfs file system
+ *
+ * @param prototxtPath  path to store model definition file
+ * @param modelPath  path to store model weights and bias
+ * @param module BigDL module to be converted
+ * @param useV2  convert to V2 layer or not
```

Comment:
Do we need to convert to V1? I think this is deprecated in the future? maybe convert to V2 is enough?